### ISSUE TITLE
TS1183 Sourcecode in declaration in monorepo

### ISSUE THREAD

**Issue by Kleywalker** [timocov/dts-bundle-generator#261]
**Bug report**

In this constellation there is sourcecode in the bundled declaration.
Both packageA und packageB are used in one monorepo.
my.d.ts(4,18): error TS1183: An implementation cannot be declared in ambient contexts.

**Input code**

package.json
```json
{
  "name": "definitionbundle",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build": "cd packages/packageA && npm run build"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "dts-bundle-generator": "^8.0.1",
    "typescript": "^5.2.2"
  },
  "workspaces": [
    "packages/packageA",
    "packages/packageB"
  ]
}
```

packages/packageA/src/package.json
```json
{
  "name": "packagea",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.ts",
  "dependencies": {
    "packageB": "1.0.0"
  },
  "scripts": {
    "build": "dts-bundle-generator -o my.d.ts src/index.ts"
  },
  "author": "",
  "license": "ISC"
}
```

packages/packageA/src/index.ts
```ts
import { ClassB } from "packageB";

class ClassA {
  fump() {
    return new ClassB();
  }
}

export { ClassA };
```

packages/packageB/src/package.json
```json
{
  "name": "packageb",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.ts",
  "devDependencies": {},
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

packages/packageB/src/index.ts
```ts
class ClassB {
  do(bla: string) {
    console.log(bla);
  }
}
export { ClassB };
```

There is only one empty tsconfig.json in packages/packageA/.

**Expected output**

```ts
// Generated by dts-bundle-generator v8.0.1

declare class ClassB {
	do(bla: string): void
}
export declare class ClassA {
	fump(): ClassB;
}

export {};
```

**Actual output**

```ts
// Generated by dts-bundle-generator v8.0.1

declare class ClassB {
	do(bla: string) {
		console.log(bla);
	}
}
export declare class ClassA {
	fump(): ClassB;
}

export {};
```

**Additional context**
I use this CLI command in packageA
```bash
dts-bundle-generator -o my.d.ts src/index.ts
```

**Comment by timocov** [comment]
Are you getting any warnings or log messages while running the tool (apart from the error while checking the output)?

**Comment by Kleywalker** [comment]
This is the full output:
```bash
dts-bundle-generator -o my.d.ts src/index.ts

Compiling input files...
WARNING: It seems that some files in the compilation still are not declaration files.
For more information see https://github.com/timocov/dts-bundle-generator/issues/53.
If you think this is a mistake, feel free to open new issue or just ignore this warning.
  /Users/stefan/code/definitionbundle/packages/packageB/src/index.ts

Processing src/index.ts
Writing src/index.ts -> my.d.ts
Checking generated files...
my.d.ts(4,18): error TS1183: An implementation cannot be declared in ambient contexts.

Error: Compiled with errors
```

**Comment by timocov** [comment]
Hey @Kleywalker can you provide a full repro please? I just tried to create files you mentioned and the file structure and it just doesn't run the tool and fails at various stages.

**Comment by Kleywalker** [comment]
Hi @timocov 😊 Have a look at this repository:
https://github.com/Kleywalker/definition-bundle
The instructions to reproduce are in the README.md.

BTW: Placing the package.json in src was a typo. Sorry for that.

**Comment by timocov** [comment]
Thanks for creating a repro @Kleywalker!

After checking what's going on there I can say that this is by design, more information about that you can find in [this issue](https://github.com/timocov/dts-bundle-generator/issues/213) (see comments https://github.com/timocov/dts-bundle-generator/issues/213#issuecomment-1163447536 and https://github.com/timocov/dts-bundle-generator/issues/213#issuecomment-1172951615) and https://github.com/timocov/dts-bundle-generator/issues/53#issuecomment-499264530.

tldr: consider compile your packages before running the tool (so you can get `d.ts` files) or use [`paths` compiler options](https://www.typescriptlang.org/tsconfig#paths) like so:

```json
{
	"compilerOptions": {
		"paths": {
			"packageB": [
				"../packageB/src"
			]
		}
	}
}
```

if you add this in `packages/packageA/tsconfig.json` it seems to work well, but please be aware of the side-effect that if you're gonna use this tsconfig for compiling the project it will compile some other packages as well, so you'd probably need to create a separate tsconfig for dts bundling (in case if you don't want to compile your packages in prior bundling).

Feel free to ask any questions.

**Comment by Kleywalker** [comment]
Using `paths` in `compilerOptions` works for me. Thanks! 😊
